internal/provider/incident_workflow_resource.go

```diff
@@ -0,0 +1,210 @@
+	return &IncidentWorkflowResource{}
+}
+
+type IncidentWorkflowResourceModel struct {
+	ID               types.String `tfsdk:"id"`
+	Name             types.String `tfsdk:"name"`
+	Folder           types.String `tfsdk:"folder"`
+	Trigger          types.String `tfsdk:"trigger"`
+	TerraformRepoURL types.String `tfsdk:"terraform_repo_url"`
+}
+
+func (r *IncidentWorkflowResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_workflow"
+}
+
+func (r *IncidentWorkflowResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: apischema.TagDocstring("Workflows V2"),
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+			"folder": schema.StringAttribute{
+				Optional: true,
+			},
+			"trigger": schema.StringAttribute{
+				Required: true,
+			},
+			"terraform_repo_url": schema.StringAttribute{
+				Optional: true,
+			},
+		},
+	}
+}
+
+func (r *IncidentWorkflowResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *IncidentWorkflowResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	payload := client.CreateWorkflowRequestBody{
+		Trigger: data.Trigger.ValueString(),
+		Workflow: client.WorkflowPayload{
+			Name:             data.Name.ValueString(),
+			TerraformRepoUrl: data.TerraformRepoURL.ValueStringPointer(),
+			OnceFor:          []string{"incident.url"},
+			ConditionGroups:  []client.ExpressionFilterOptsPayloadV2{},
+			Steps:            []client.StepConfigPayload{},
+			Expressions:      []client.ExpressionPayloadV2{},
+			RunsOnIncidents:  "newly_created",
+			IsDraft:          true,
+		},
+	}
+	if folder := data.Folder.ValueString(); folder != "" {
+		payload.Folder = &folder
+	}
+
+	result, err := r.client.WorkflowsV2CreateWorkflowWithResponse(ctx, payload)
+	if err == nil && result.StatusCode() >= 400 {
+		err = fmt.Errorf(string(result.Body))
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create workflow, got error: %s", err))
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("created a workflow resource with id=%s", result.JSON201.Workflow.Id))
+	data = r.buildModel(result.JSON201.Workflow)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *IncidentWorkflowResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state *IncidentWorkflowResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var data *IncidentWorkflowResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
```

Comment:
read from state to find the `ID`, but from the _plan_ to find the payload. we could also consider merging non-empty fields from the plan onto the state and using the merged payload in the JSON request body.. this isn't set up for now